R/src/feather-read.cpp

```diff
@@ -47,17 +47,55 @@ List metadataFeather(const std::string& path) {
   return out;
 }
 
+
 // [[Rcpp::export]]
-List readFeather(const std::string& path) {
-  auto table = openFeatherTable(path);
+XPtr<feather::TableReader> openFeather(const std::string& path) {
+  return XPtr<TableReader>(openFeatherTable(path).release(), true);
+}
 
-  int n = table->num_columns(), p = table->num_rows();
-  List out(n), names(n);
+
+// [[Rcpp::export]]
+double colsFeather(List feather) {
+  auto table = Rcpp::as<XPtr<TableReader> >(feather["ptr"]);
```

Comment:
I wonder if it would be nicer for these to functions to have input as `XPtr<TableReader>` (i.e. do the extraction on the R side)